File: mobile/ios/CentralOperacionalApp/Views/Components/BottomTabBar.swift
import SwiftUI

/// Navegação inferior fixa (Central | Pedidos | Cozinha | Bar | Caixa) —
/// custom (não `TabView`) para reproduzir o indicador de aba ativa do web.
struct BottomTabBar: View {
    @Binding var selection: ModuleID
    let onSelect: (ModuleID) -> Void

    private let tabs: [ModuleID] = [.central, .pedidos, .cozinha, .bar, .caixa]

    var body: some View {
        HStack(spacing: 0) {
            ForEach(tabs) { tab in
                let isActive = tab == selection
                Button {
                    selection = tab
                    onSelect(tab)
                } label: {
                    VStack(spacing: 4) {
                        Capsule()
                            .fill(isActive ? Color.ppPrimary : .clear)
                            .frame(width: 28, height: 3)

                        Image(systemName: tab.systemImage)
                            .font(.system(size: 20, weight: isActive ? .semibold : .regular))
                        Text(tab.title)
                            .font(PPFont.tabLabel)
                    }
                    .foregroundStyle(isActive ? Color.ppPrimary : Color.ppTextMuted)
                    .frame(maxWidth: .infinity)
                    .padding(.top, 6)
                }
                .buttonStyle(.plain)
                .accessibilityAddTraits(isActive ? [.isButton, .isSelected] : .isButton)
            }
        }
        .padding(.bottom, 2)
        .background(
            Color.ppSurface
                .overlay(Divider(), alignment: .top)
                .ignoresSafeArea(edges: .bottom)
        )
    }
}

#Preview {
    BottomTabBar(selection: .constant(.central)) { _ in }
}

File: mobile/android/app/src/main/java/com/pedidoprime/centraloperacional/ui/components/BottomTabBar.kt
package com.pedidoprime.centraloperacional.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.weight
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.semantics.role
import androidx.compose.ui.semantics.selected
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.unit.dp
import com.pedidoprime.centraloperacional.model.ModuleId
import com.pedidoprime.centraloperacional.ui.theme.PPFont
import com.pedidoprime.centraloperacional.ui.theme.icon
import com.pedidoprime.centraloperacional.ui.theme.title

/** Navegação inferior fixa (Central | Pedidos | Cozinha | Bar | Caixa). */
@Composable
fun BottomTabBar(
    selectedTab: ModuleId,
    onSelect: (ModuleId) -> Unit,
    surfaceColor: Color,
    borderColor: Color,
    activeColor: Color,
    inactiveColor: Color,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .background(surfaceColor)
            .border(width = 1.dp, color = borderColor)
            .padding(bottom = 2.dp, top = 6.dp)
    ) {
        ModuleId.TABS.forEach { tab ->
            val isActive = tab == selectedTab
            val tint = if (isActive) activeColor else inactiveColor
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(4.dp),
                modifier = Modifier
                    .weight(1f)
                    .clickable { onSelect(tab) }
                    .padding(vertical = 4.dp)
                    .semantics {
                        role = Role.Tab
                        selected = isActive
                    }
            ) {
                Column(
                    modifier = Modifier
                        .size(width = 28.dp, height = 3.dp)
                        .background(if (isActive) activeColor else Color.Transparent, RoundedCornerShape(50))
                ) {}
                Icon(tab.icon, contentDescription = null, tint = tint)
                Text(tab.title(), color = tint, style = PPFont.tabLabel)
            }
        }
    }
}
